Specification/4._CAN_bus_transport_layer.md

```diff
@@ -118,69 +114,69 @@ If different values are used, they must be explicitly documented.
-Transfer descriptor is a set of properties that identify a particular set of transfers that originate from
+For *message transfers* and *service request transfers* the ID value should be computed as described below.
+For *service response transfers* this value must be directly copied from the corresponding service request
+transfer.
+
+The logic to compute the *Transfer ID* relies on the concept of *transfer descriptor*.
+A transfer descriptor is a set of properties that identify a particular set of transfers that originate from
 the same node, share the same data type ID and the same type.
 The properties that constitute a transfer descriptor are listed below:
 
-* Transfer type (message broadcasting, service request, etc.)
+* Transfer type (message broadcast, service request, etc.)
 * Data type ID
 * Source node ID
 * Destination node ID (only for unicast transfers)
 
 Every node that needs to publish a transfer must maintain the mapping from transfer descriptors to transfer ID.
-This mapping will be referred to as *transfer ID map*.
+This mapping is referred to as the *transfer ID map*.
 
 Whenever a node needs to emit a transfer, it will query its transfer ID map for the appropriate transfer descriptor.
 If the map does not contain such entry, a new entry will be created with transfer ID initialized to zero.
 The node will use the current value of transfer ID from the map for the transfer, and then the value stored in the map
 will be incremented by one.
 When the stored transfer ID exceeds its maximum value, it will roll over to zero.
 
-It is expected that some nodes will need to publish certain transfers aperiodically or ad-hoc, therefore creating
+It is expected that some nodes will need to publish certain transfers aperiodically or on an ad-hoc basis, thereby creating
 unused entries in the transfer ID map.
 In order to avoid keeping unused entries in the map, the nodes are allowed, but not required,
 to remove entries from the map that were not used for at least 2 seconds.
 Therefore, it is possible that a node may publish a transfer with an out-of-order transfer ID value, if previous
-transfer of the same descriptor has been published more than 2 seconds ago.
+transfer of the same descriptor has been published more than 2 seconds previously.
 
 ### Payload decomposition
 
 #### Single frame transfer
 
-If size of the entire transfer payload does not exceed the space available for payload in a single CAN frame,
+If the size of the entire transfer payload does not exceed the space available for payload in a single CAN frame,
 the whole transfer will be contained in one CAN frame. Such transfer is called a *single-frame transfer*.
 
 Single frame transfers are more efficient than multi-frame transfers in terms of throughput and latency,
-therefore it is advised to avoid the latter when appropriate.
+therefore it is advised to avoid the latter where possible.
```

Comment:
@pavel-kirienko Not sure about this change - generally if I see "when appropriate" I want to have guidance one when that might be. While "where possible" pretty much says - always try to do this.

Reply:
Okay, this is a good change.